driver-core/src/main/com/mongodb/connection/ByteBufferBsonOutput.java

```diff
@@ -177,7 +177,9 @@ public void truncateToPosition(final int newPosition) {
     @Override
     public void close() {
         for (final ByteBuf cur : bufferList) {
-            cur.release();
+            while(cur.getReferenceCount() > 0) {
```

Comment:
Yes we aren't releasing it, because we aren't explicitly retaining it.  Everytime we duplicate the underlying ByteBufs we retain a copy. So calling `getBuffers()` causes netty's ref count to increase.   I'll rework and propose an alternative solution.